packages/shared/src/types/Plugin.ts

```diff
@@ -33,7 +33,7 @@ export interface RspressPlugin {
   /**
    * Rsbuild config.
    */
-  builderConfig?: RsbuildConfig;
+  builderConfig?: RsbuildConfig | ((config: RsbuildConfig, isSSR: boolean) => void | RsbuildConfig);
```

Comment:
the second param should be an object, making it  easier to add more params in the future.

Reply:
The "config" hook supports function type, maybe we should use it to modify builderConfig.

Reply:
The config hook don't have internalRsbuildConfig, derived from the rspress configuration.
The config hook still process rspress config,.

Reply:
Sure, if you need to process the builtin Rsbuild config, you can write an Rsbuild plugin.